client/state/selectors/get-network-sites.js

```diff
@@ -0,0 +1,38 @@
+/**
+ * External dependencies
+ */
+import { get } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import createSelector from 'lib/create-selector';
+import { isMainSiteOf } from 'state/selectors';
+import {
+	getSite,
+	isJetpackSiteMainNetworkSite
+} from 'state/sites/selectors';
+
+/**
+ * Returns the secondary sites of a given site
+ * Returns null if main site is not found
+ *
+ * @param  {Object}  state     Global state tree
+ * @param  {Number}  siteId    The ID of the main site for which we're retrieving the secondary sites
+ * @return {?Array}            Array of secondary sites
+ */
+export default createSelector(
+	( state, siteId ) => {
+		const isMainNetworkSite = isJetpackSiteMainNetworkSite( state, siteId );
+		if ( isMainNetworkSite == null ) {
+			return null;
+		}
+		if ( ! isMainNetworkSite ) {
```

Comment:
I think this case is confusing. I'd probably expect it to return `null` if the site is not the main network site. Because this one can easily be confused with the case when there is a main network site without any connected secondary sites.